main.py

```diff
@@ -49,11 +65,25 @@ def showMap(lat, lon):
         'marker': f'lonlat:{lon},{lat};color:#ff0000;size:medium',
         'apiKey': GEOAPIFY_API_KEY
     }    
-    full_url = f'{url}{urllib.parse.urlencode(params)}'
-    webbrowser.open(full_url)
+    fullUrl = f'{url}{urllib.parse.urlencode(params)}'
+    
+    return fullUrl
+
+@app.route('/', methods=['GET', 'POST'])
+def index():
+    cityName = getRandomCityName(POPULATION_THRESHHOLD)
+    cityCoords = getCityCoordinatesByName(cityName)
+    city = City(cityName, cityCoords)
+    mapOfCity = showMap(city)
+    
+    return render_template('index.html', mapOfCity=mapOfCity)
     
 if __name__ == '__main__':
-    city = randomCity()
-    lat, lon = searchCity(city)
-    showMap(lat, lon)
-        
+    app.secret_key = os.urandom(24)
```

Comment:
Any reason to make a new key everytime as opposed to generating one separately and including it in your `.env` similar to what the docs do? https://flask.palletsprojects.com/en/2.0.x/config/#SECRET_KEY

Reply:
Hadn't thought about storing it as such. I'll go about that.